src/js/actions/export.js

```diff
@@ -727,32 +727,36 @@ define(function (require, exports) {
         return this.transfer(promptForFolder)
             .bind(this)
             .then(function (baseDir) {
-                _lastFolderPath = baseDir;
-
-                var layerIdList = collection.pluck(layersList, "id");
-                return _setAssetsRequested.call(this, document.id, layerIdList);
-            })
-            .then(_setServiceBusy.bind(this, true))
-            .return(quickAddPromise)
-            .then(function () {
-                // fetch documentExports anew, in case quick-add added any assets
-                var documentExports = this.flux.stores.export.getDocumentExports(documentID, true);
-
-                // Iterate over the layers, find the associated export assets, and export them
-                var exportList = layersList.flatMap(function (layer) {
-                    var prefix = prefixMap && prefixMap.get(layer.id);
-
-                    return documentExports.getLayerExports(layer.id)
-                        .map(function (asset, index) {
-                            return _exportAsset.call(this, document, layer, index, asset, _lastFolderPath, prefix);
-                        }, this);
-                }, this);
-
-                _batchExports.call(this, exportList);
-                return Promise.resolve();
-            })
-            .catch(ExportService.CancelPromptError, function () {
-                return Promise.resolve();
+                if (!baseDir) {
+                    return Promise.resolve();
+                } else {
```

Comment:
suggestion: halting a promise chain in the middle could be painful; an easy way might be throw an exception (instead of returning a resolved promise) and add a `catch` at the end of the chain; in this way,  you can keep most of your code unchanged, and handle the edge cases nicely.